### PR TITLE
Cleans up some HL library code that inappropriately returns htri_t values cast to herr_t

### PR DESCRIPTION
**Pull request by derobins**

### CODE REVIEW

File: hl/src/H5LTprivate.h [hunk 1]
```diff
@@ -20,23 +20,12 @@
...
- */
-
 H5_HLDLL herr_t H5LT_get_attribute_disk(hid_t obj_id, const char *attr_name, void *data);
-
 H5_HLDLL herr_t H5LT_set_attribute_numerical(hid_t loc_id, const char *obj_name, const char *attr_name,
                                              size_t size, hid_t type_id, const void *data);
-
 H5_HLDLL herr_t H5LT_set_attribute_string(hid_t dset_id, const char *name, const char *buf);
-
-H5_HLDLL herr_t H5LT_find_attribute(hid_t loc_id, const char *name);
```
Comment (derobins) [inline hl/src/H5LTprivate.h:35]: The only real change in this file was that this function was removed.

File: hl/src/H5LT.c [hunk 2]
```diff
@@ -1858,49 +1855,24 @@ H5LTset_attribute_double(hid_t loc_id, const char *obj_name, const char *attr_na
...
- *  Calls the private version of the function
+ * TODO:     Overloading herr_t is not a great idea. This function either
+ *           needs to be rewritten to take a Boolean out parameter in
+ *           HDF5 2.0 or possibly even eliminated entirely as it simply
+ *           wraps H5Aexists.
  *
+ * Return:   An htri_t value cast to herr_t
+ *              Exists:         Positive
+ *              Does not exist: 0
+ *              Error:          Negative
```
Comment (derobins) [inline hl/src/H5LT.c:1869]: htri_t is an abomination. I'm using how it's described in H5public.h.

File: hl/src/H5LT.c [hunk 3]
```diff
@@ -1858,49 +1855,24 @@ H5LTset_attribute_double(hid_t loc_id, const char *obj_name, const char *attr_na
...
- *
- *-------------------------------------------------------------------------
- */
-herr_t
-H5LT_find_attribute(hid_t loc_id, const char *attr_name)
-{
-    htri_t attr_exists = H5Aexists(loc_id, attr_name);
-    return (attr_exists < 0) ? (herr_t)-1 : (attr_exists) ? (herr_t)1 : (herr_t)0;
+    return (herr_t)H5Aexists(loc_id, attr_name);
 }
```
Comment (derobins) [inline hl/src/H5LT.c:1876]: This function is completely useless so I just use H5Aexists directly throughout the HL library instead.

File: hl/src/H5LT.c (hunk 2 above)
Comment (qkoziol) [reply hl/src/H5LT.c:1869]: Yeah, not certain what we were thinking there...

**Review by qkoziol — APPROVED** [review]

File: hl/src/H5LT.c [hunk 4]
```diff
@@ -1858,49 +1855,24 @@ H5LTset_attribute_double(hid_t loc_id, const char *obj_name, const char *attr_na
...
- *
- * Programmer: Pedro Vicente
- *
- * Date: May 17, 2006
+ * Purpose:  Checks if an attribute named attr_name exists attached to
+ *           the object loc_id
  *
- * Comments:
- *  Calls the private version of the function
+ * TODO:     Overloading herr_t is not a great idea. This function either
```
Comment (hyoklee) [inline hl/src/H5LT.c:1861]: Use  `\todo`:

https://www.doxygen.nl/manual/commands.html#cmdtodo

**Review by hyoklee — APPROVED** [review]

**Review by byrnHDF — APPROVED** [review]

**Pull request merged**